YoutubeExplode/YoutubeClient.cs

```diff
@@ -186,6 +186,44 @@ public YoutubeClient()
         }
 
         /// <summary>
+        /// Gets all videos in a given playlist.
+        /// This method will does one GET request per every 20 videos in the playlist.
+        /// 
+        /// This also works for special playlists like the Watch Later playlist at https://www.youtube.com/playlist?list=WL,
+        /// but the _requestServices has to mimic a logged in user.
+        /// Same for other non public playlists.
+        /// </summary>
+        public async Task<List<string>> GetVideoIdsInPlaylistAsync(string playlistId)
+        {
+            var videoIds = new List<string>();
+
+            //We use a youtube api that is also used by the mobile version of the youbute page (m.youtube.com).
+            //For this to work we need to add &ajax=1&tsp=1&app=m to our playlist url
+            var baseUrl = $"https://m.youtube.com/playlist?list={playlistId}&ajax=1&tsp=1&app=m";
+            var nextUrl = baseUrl;
+            while (nextUrl != null)
```

Comment:
What do you think of using recursion instead of a while loop? Is there a risk of stack overflow? I think it can make the code cleaner.